Server/Server.Core/Server.Core.Static/ArgParser.hpp

```diff
@@ -124,12 +127,34 @@ class ArgParser
     * @details Pairs arguments with their values.
     * @param std::string const parameter(argument name).
     */
+
     std::pair<std::string, std::string> getPair(std::string const& parameter) const
     {
         std::pair<std::string, std::string> args = 
             std::make_pair(parameter, _argParser.get(parameter));
         return args;
     }
+    
+    /**
+    * @brief Used to generate Settings class object
+    * @details Fills Settings class object with the current parameter values and returns it
+    */
+    std::unique_ptr<Server::Builder::Settings> getSettings() const
+    {
+        auto getPair = [this](const std::string& str)
+                       {
+                           return std::pair<std::string, std::string>(str, _argParser.get(str));
+                       };
+
+        auto result = std::make_unique<Server::Builder::Settings>();
+        (*result).setValue(getPair(ParamType::ServerPort))
```

Comment:
Why not ->?

Reply:
This is not  an attractive way of chaining commands with "->" in the first step and "." on other steps. And this syntax does not contradict generally accepted coding norms.